### PR TITLE
Tests for  TimeZone.localizedName

### PR DESCRIPTION
**Pull request by saiHemak**

### CODE REVIEW

File: TestFoundation/TestNSTimeZone.swift [hunk 1]
```diff
@@ -117,6 +117,20 @@ class TestNSTimeZone: XCTestCase {
         let known = NSTimeZone.knownTimeZoneNames
         XCTAssertNotEqual([], known, "known time zone names not expected to be empty")
     }
+    
+    func test_localizedName(){
+        let initialTimeZone = NSTimeZone.default
+        NSTimeZone.default = TimeZone(identifier: "America/New_York")!
+        let defaultTimezone = NSTimeZone.default
```
Comment (xwu) [inline TestFoundation/TestNSTimeZone.swift]: Nit: for consistent capitalization, this should be `defaultTimeZone`

File: TestFoundation/TestNSTimeZone.swift [hunk 2]
```diff
@@ -117,6 +117,20 @@ class TestNSTimeZone: XCTestCase {
...
+        let localeL = Locale(identifier: "en_US")
+        XCTAssertEqual(defaultTimezone.localizedName(for: NSTimeZone.NameStyle.standard, locale: localeL), "Eastern Standard Time")
+        XCTAssertEqual(defaultTimezone.localizedName(for: NSTimeZone.NameStyle.shortStandard, locale: localeL), "EST")
+        XCTAssertEqual(defaultTimezone.localizedName(for: NSTimeZone.NameStyle.generic, locale: localeL), "Eastern Time")
+        XCTAssertEqual(defaultTimezone.localizedName(for: NSTimeZone.NameStyle.daylightSaving, locale: localeL), "Eastern Daylight Time")
+        XCTAssertEqual(defaultTimezone.localizedName(for: NSTimeZone.NameStyle.shortDaylightSaving, locale: localeL), "EDT")
+        XCTAssertEqual(defaultTimezone.localizedName(for: NSTimeZone.NameStyle.shortGeneric, locale: localeL), "ET")
+        NSTimeZone.default = initialTimeZone //reset the Timezone
```
Comment (xwu) [inline TestFoundation/TestNSTimeZone.swift]: Even more of a nit: same thing in the comment :)

File: TestFoundation/TestNSTimeZone.swift [hunk 3]
```diff
@@ -117,6 +117,20 @@ class TestNSTimeZone: XCTestCase {
         let known = NSTimeZone.knownTimeZoneNames
         XCTAssertNotEqual([], known, "known time zone names not expected to be empty")
     }
+    
+    func test_localizedName(){
```
Comment (itaiferber) [inline TestFoundation/TestNSTimeZone.swift]: I've got a few nits myself before I'm ready to merge, if you don't mind. `test_localizedName(){` → `test_localizedName() {`

File: TestFoundation/TestNSTimeZone.swift (hunk 2 above)
Comment (itaiferber) [inline TestFoundation/TestNSTimeZone.swift]: Additionally:
* Can we change `localeL` to just `locale`?
* Can we drop `NSTimeZone.NameStyle` so these cases just read `defaultTimeZone.localizedName(for: .standard, locale: locale), ...`?

**Comment by saiHemak** [comment]
@itaiferber  Thanks for the review.I have addressed your comments ..

**Comment by itaiferber** [comment]
Looks good to me!

**Comment by itaiferber** [comment]
@swift-ci Please test

**Comment by itaiferber** [comment]
@parkera Should this be submitted against swift-3.0-branch?

**Comment by parkera** [comment]
Let's leave it on master; no need for pure testing code on the swift 3 branch.

**Pull request merged**